google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/AppEngineDeploymentRunConfigurationEditor.java

```diff
@@ -363,6 +367,22 @@ private String getDisplayableVersion() {
   }
 
   private void validateConfiguration() throws ConfigurationException {
+    CloudSdkService sdkService = CloudSdkService.getInstance();
+
+    // Perform cloud sdk validation
+    Path cloudSdkPath = sdkService.getSdkHomePath();
+    Set<CloudSdkValidationResult> results = sdkService.validateCloudSdk(cloudSdkPath);
```

Comment:
Can the java components check (see the last else if) also be encapsulated into your new validation model?